### ISSUE TITLE
Waiting for plugin host to spawn

### ISSUE THREAD

**Issue by romgrk** [neovim/neovim#3485]
Hello,

This might be an implementation error on my part, but is there any plan to have an `rpcwait` function, analogous to `jobwait`?
Currently, I have an implementation of plugin host, but starting it with `rpcstart` makes it miss the first request if it's sent ASAP. Thus, calling `:UpdateRemotePlugins` freezes everything, as the first request (`specs`, ... ) is missed. 
The current workaround I have uses `sleep`, and doesn't feel clean.

**Comment by tarruda** [comment]
`rpcrequest` blocks until a response is received, you can call it with the id returned from `rpcstart`.

**Comment by romgrk** [comment]
Yes, that is not the problem (or it kinda is but...)

Let's suppose a plugin host registration sequence:

| stack | host |
| --- | --- |
| `:UpdateRemotePlugins` |  |
| => `remote#host#register('myhost-registration-clone')` |  |
| ==> `MyHostRequireFunction(...)`                                |  |
| ==> returns `rpcstart(...)`                                                      | *host spawning* |
| => `call rpcrequest(channel_id, 'specs', [...])`                  | *host still spawning* |
| *block* |  |

**Comment by fdinoff** [comment]
Why is the plugin host losing notification/requests when it is sent immediately after it started? I would have thought that the plugin host would have left it on stdin or the pipe so that it would be processed eventually. It might take some time for it to react to a notification but it should be handled eventually.

**Comment by romgrk** [comment]
Yes, I'll close, it's probably an implementation error on my side. 
Actually it uses the node-client as first parser, but I guess it loses the first request while doing its api parsing, or something like that. I'll give it a try with my own msgpack handler.
